#! /bin/sh -e

# DP: Description: Optimizes the build for sparcv8
# DP: Author: Ben Collins
# DP: Upstream status: Debian-Specific
# DP: Status Details:
# DP: Date: June 17, 2003

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- configure~	2003-06-17 09:24:52.000000000 -0400
+++ configure	2003-06-17 09:25:19.000000000 -0400
@@ -1796,6 +1796,11 @@
   machine="powerpc"
 fi
 
+# Force v8
+if test "$machine" = sparc; then
+   machine="sparcv8"
+fi
+
 # Braindead PowerPC box with absolutely no FPU.
 case "$machine-$host_os" in
   powerpc*-*soft)
--- configure.in~	2003-06-17 09:24:55.000000000 -0400
+++ configure.in	2003-06-17 09:25:29.000000000 -0400
@@ -337,6 +337,11 @@
   machine="powerpc"
 fi
 
+# Force v8
+if test "$machine" = sparc; then
+   machine="sparcv8"
+fi
+
 # Braindead PowerPC box with absolutely no FPU.
 case "$machine-$host_os" in
   powerpc*-*soft)
